partcad/src/partcad/project.py

```
@@ -1427,15 +1273,31 @@
                     yaml.dump(config, fp)
                     fp.close()
 
-    def test(self):
+    async def _run_test_async(self, ctx, tests: list, use_wrapper: bool = False) -> list[asyncio.Task]:
+        tasks = []
+        test_method = "test_log_wrapper" if use_wrapper else "test"
         for interface in self.interfaces.values():
-            interface.test()
+            tasks.append(asyncio.create_task(interface.test_async()))
         for sketch in self.sketches.values():
```

Ruff (0.8.2)

Import `sketch` from line 24 shadowed by loop variable

(F402)